### PR TITLE
ARTEMIS-4743 Split lines on ./artemis queue stat for large names

### PR DESCRIPTION
**Pull request by clebertsuconic**

### CODE REVIEW

**Comment by jbertram** [comment]
The headings of most columns are being wrapped, e.g.:
```
$ ./artemis queue stat --maxColumnSize=30
Connection brokerURL = tcp://localhost:61616
|NAME                          |ADDRESS                       |CONSUMER_C|MESSAGE_CO|MESSAGES_A|DELIVERING|MESSAGES_A|SCHEDULED_|ROUTING_TY|
|                              |                              |OUNT      |UNT       |DDED      |_COUNT    |CKED      |COUNT     |PE        |
|$sys.mqtt.sessions            |$sys.mqtt.sessions            |0         |0         |0         |0         |0         |0         |ANYCAST   |
|DLQ                           |DLQ                           |0         |0         |0         |0         |0         |0         |ANYCAST   |
|ExpiryQueue                   |ExpiryQueue                   |0         |0         |0         |0         |0         |0         |ANYCAST   |
|activemq.management.c9ac407e-f|activemq.management.c9ac407e-f|1         |1         |1         |1         |0         |0         |MULTICAST |
|9a3-48c0-a4c4-c760a01e9170    |9a3-48c0-a4c4-c760a01e9170    |          |          |          |          |          |          |          |
|activemq.management.e9fe512e-3|activemq.management.e9fe512e-3|1         |0         |0         |0         |0         |0         |MULTICAST |
|77e-415d-ba37-a9cd2e12e0f8    |77e-415d-ba37-a9cd2e12e0f8    |          |          |          |          |          |          |          |
|activemq.management.ea027bf5-5|activemq.management.ea027bf5-5|1         |1         |1         |1         |0         |0         |MULTICAST |
|a8d-427c-a745-78bf20302274    |a8d-427c-a745-78bf20302274    |          |          |          |          |          |          |          |
|fo0000000000000000000000000000|barrrrrrrrrrrrrrrrrrrrrrrrrrrr|0         |0         |0         |0         |0         |0         |ANYCAST   |
|0000000000                    |rrrrrrrrrr                    |          |          |          |          |          |          |          |
```

**Comment by erwindon** [comment]
some people rely on this output to be useable in tools like AWK or other processors. that would cause problems when header and/or data are wrapped. is there at least an option that prevents all wrapping? (at the expense of very wide tables). In the example from Justin, the numeric columns are smaller than the maxColumnSize, yet they still wrap the header.

**Comment by clebertsuconic** [comment]
@erwindon I would keep this output as visual... you have lots of APIs. I wouldn't commit to the report as an API.

**Comment by clebertsuconic** [comment]
@erwindon you can actually get the same report with a JSON/Jolokia call and do the processing in any language.


I don't want to stick to a format because of any possible usage.


As a matter of fact I will have to add a new column to this report, about the queue being internal or not.

**Comment by clebertsuconic** [comment]
@jbertram :  

```
 bin % ./artemis queue stat
Connection brokerURL = tcp://localhost:61616
|NAME                     |ADDRESS                  |CONSUMER|MESSAGE|MESSAGES|DELIVERING|MESSAGES|SCHEDULED|ROUTING  |
|                         |                         |COUNT   |COUNT  |ADDED   |COUNT     |ACKED   |COUNT    |TYPE     |
|$ACTIVEMQ_ARTEMIS_MIRROR_|$ACTIVEMQ_ARTEMIS_MIRROR_|0       |9      |9       |0         |0       |0        |ANYCAST  |
|mirror                   |mirror                   |        |       |        |          |        |         |         |
|$sys.mqtt.sessions       |$sys.mqtt.sessions       |0       |0      |0       |0         |0       |0        |ANYCAST  |
|DLQ                      |DLQ                      |0       |0      |0       |0         |0       |0        |ANYCAST  |
|ExpiryQueue              |ExpiryQueue              |0       |0      |0       |0         |0       |0        |ANYCAST  |
|activemq.management.1fa0d|activemq.management.1fa0d|1       |1      |1       |1         |0       |0        |MULTICAST|
|c04-dca0-4c64-a419-aa74e6|c04-dca0-4c64-a419-aa74e6|        |       |        |          |        |         |         |
|0bd5f2                   |0bd5f2                   |        |       |        |          |        |         |         |
|activemq.management.d1c1f|activemq.management.d1c1f|1       |1      |1       |1         |0       |0        |MULTICAST|
|ad5-b2f2-482a-b2db-08027c|ad5-b2f2-482a-b2db-08027c|        |       |        |          |        |         |         |
|78df2b                   |78df2b                   |        |       |        |          |        |         |         |
|activemq.management.d6e49|activemq.management.d6e49|1       |0      |0       |0         |0       |0        |MULTICAST|
|20e-8bce-4a59-8b8e-65372c|20e-8bce-4a59-8b8e-65372c|        |       |        |          |        |         |         |
|043b02                   |043b02                   |        |       |        |          |        |         |         |
|myQueue                  |myQueue                  |0       |0      |0       |0         |0       |0        |ANYCAST  |
```

**Comment by jbertram** [comment]
I'm not trying to nitpick here, but I wonder if the wrapped queue and address names would be more readable if they were indented a few spaces, e.g.
```
% ./artemis queue stat
Connection brokerURL = tcp://localhost:61616
|NAME                     |ADDRESS                  |CONSUMER|MESSAGE|MESSAGES|DELIVERING|MESSAGES|SCHEDULED|ROUTING  |
|                         |                         |COUNT   |COUNT  |ADDED   |COUNT     |ACKED   |COUNT    |TYPE     |
|$ACTIVEMQ_ARTEMIS_MIRROR_|$ACTIVEMQ_ARTEMIS_MIRROR_|0       |9      |9       |0         |0       |0        |ANYCAST  |
|mirror                   |mirror                   |        |       |        |          |        |         |         |
|$sys.mqtt.sessions       |$sys.mqtt.sessions       |0       |0      |0       |0         |0       |0        |ANYCAST  |
|DLQ                      |DLQ                      |0       |0      |0       |0         |0       |0        |ANYCAST  |
|ExpiryQueue              |ExpiryQueue              |0       |0      |0       |0         |0       |0        |ANYCAST  |
|activemq.management.1fa0d|activemq.management.1fa0d|1       |1      |1       |1         |0       |0        |MULTICAST|
|  c04-dca0-4c64-a419-aa74|  c04-dca0-4c64-a419-aa74|        |       |        |          |        |         |         |
|  e60bd5f2               |  e60bd5f2               |        |       |        |          |        |         |         |
|activemq.management.d1c1f|activemq.management.d1c1f|1       |1      |1       |1         |0       |0        |MULTICAST|
|  ad5-b2f2-482a-b2db-0802|  ad5-b2f2-482a-b2db-0802|        |       |        |          |        |         |         |
|  7c78df2b               |  7c78df2b               |        |       |        |          |        |         |         |
|activemq.management.d6e49|activemq.management.d6e49|1       |0      |0       |0         |0       |0        |MULTICAST|
|  20e-8bce-4a59-8b8e-6537|  20e-8bce-4a59-8b8e-6537|        |       |        |          |        |         |         |
|  2c043b02               |  2c043b02               |        |       |        |          |        |         |         |
|myQueue                  |myQueue                  |0       |0      |0       |0         |0       |0        |ANYCAST  |
```

**Comment by erwindon** [comment]
@clebertsuconic
Thanks for the replies!
Personally I'm using both JMX and metrics, so I'm fine with this kind of improvements.
But I also remember that I used this years ago before I mastered the JMX api :-)
thus the above little check...

**Comment by clebertsuconic** [comment]
@erwindon you can use curl with some jolokia-fu as well :) 


Don't ask me how to use it.. every time I needed it I figured out how to use it to then forget again :) 

But it should be easy enough.

**Comment by clebertsuconic** [comment]
@jbertram last iteraction... notice I renamed some of the titles...


```
bin % ./artemis queue stat
Connection brokerURL = tcp://localhost:61616
|NAME                     |ADDRESS                  |CONSUMERS|MESSAGES|MESSAGES|DELIVERING|ACKED|SCHEDULED|ROUTING  |INTERNAL|
|                         |                         |         |COUNT   |ADDED   |          |     |         |         |        |
|$ACTIVEMQ_ARTEMIS_MIRROR_|$ACTIVEMQ_ARTEMIS_MIRROR_|1        |0       |10      |0         |10   |0        |ANYCAST  |true    |
|  mirror                 |  mirror                 |         |        |        |          |     |         |         |        |
|$sys.mqtt.sessions       |$sys.mqtt.sessions       |0        |0       |0       |0         |0    |0        |ANYCAST  |true    |
|DLQ                      |DLQ                      |0        |0       |0       |0         |0    |0        |ANYCAST  |false   |
|ExpiryQueue              |ExpiryQueue              |0        |0       |0       |0         |0    |0        |ANYCAST  |false   |
|activemq.management.60de5|activemq.management.60de5|1        |0       |0       |0         |0    |0        |MULTICAST|false   |
|  b0e-ae80-4b75-bf92-10af|  b0e-ae80-4b75-bf92-10af|         |        |        |          |     |         |         |        |
|  9e724fc4               |  9e724fc4               |         |        |        |          |     |         |         |        |
|activemq.management.6be36|activemq.management.6be36|1        |1       |1       |1         |0    |0        |MULTICAST|false   |
|  de9-a7ec-411a-bd53-3411|  de9-a7ec-411a-bd53-3411|         |        |        |          |     |         |         |        |
|  cee197a3               |  cee197a3               |         |        |        |          |     |         |         |        |
|activemq.management.812a5|activemq.management.812a5|1        |1       |1       |1         |0    |0        |MULTICAST|false   |
|  363-f5bc-4517-99d2-9af2|  363-f5bc-4517-99d2-9af2|         |        |        |          |     |         |         |        |
|  4c725f4b               |  4c725f4b               |         |        |        |          |     |         |         |        |
|myQueue                  |myQueue                  |0        |0       |0       |0         |0    |0        |ANYCAST  |false   |

```

**Comment by clebertsuconic** [comment]
@erwindon but I will add a note to the release notes pointing the change

File: docs/user-manual/versions.adoc [hunk 1]
```diff
@@ -32,6 +32,8 @@ However, connections will no longer be pooled regardless of the configuration.
   ** This was done in an attempt to allow administrators to act when things are not working as expected, to get metrics on these objects and allow more transparency for the broker.
   ** this includes all Openwire Advisor queues and addresses, MQTT internal objects, Cluster Store and Forward (SNF) Queues, Mirror SNF.
   ** You may want to revisit authorizations if you mean to control access to certain users on the web console.
+* The CLI operation `./artemis queue stat` has its output improved and updated. If you parse the previous output you will see differences in the output.
```
Comment (clebertsuconic) [inline docs/user-manual/versions.adoc:35]: @erwindon as promised ^^^
let me know if you can think of a better text here or anywhere else

**Comment by clebertsuconic** [comment]
this is the last updated:

```
Connection brokerURL = tcp://localhost:61616
|NAME              |ADDRESS           |CONSUMER|MESSAGE|MESSAGES|DELIVERING|MESSAGES|SCHEDULED|ROUTING|INTERNAL|
|                  |                  | COUNT  | COUNT | ADDED  |  COUNT   | ACKED  |  COUNT  | TYPE  |        |
|$sys.mqtt.sessions|$sys.mqtt.sessions|   0    |   0   |   0    |    0     |   0    |    0    |ANYCAST|  true  |
|DLQ               |DLQ               |   0    |   0   |   0    |    0     |   0    |    0    |ANYCAST| false  |
|ExpiryQueue       |ExpiryQueue       |   0    |   0   |   0    |    0     |   0    |    0    |ANYCAST| false  |
|TEST              |TEST              |   0    |   0   |   0    |    0     |   0    |    0    |ANYCAST| false  |
```

**Comment by clebertsuconic** [comment]
same thing, but including management queues:

```
bin % ./artemis queue stat --include-management
Connection brokerURL = tcp://localhost:61616
|NAME                     |ADDRESS                  |CONSUMER|MESSAGE|MESSAGES|DELIVERING|MESSAGES|SCHEDULED| ROUTING |INTERNAL|
|                         |                         | COUNT  | COUNT | ADDED  |  COUNT   | ACKED  |  COUNT  |  TYPE   |        |
|$sys.mqtt.sessions       |$sys.mqtt.sessions       |   0    |   0   |   0    |    0     |   0    |    0    | ANYCAST |  true  |
|DLQ                      |DLQ                      |   0    |   0   |   0    |    0     |   0    |    0    | ANYCAST | false  |
|ExpiryQueue              |ExpiryQueue              |   0    |   0   |   0    |    0     |   0    |    0    | ANYCAST | false  |
|TEST                     |TEST                     |   0    |   0   |   0    |    0     |   0    |    0    | ANYCAST | false  |
|activemq.management.0fc9e|activemq.management.0fc9e|   1    |   1   |   1    |    1     |   0    |    0    |MULTICAST| false  |
|  b03-3934-4f2a-8eec-5202|  b03-3934-4f2a-8eec-5202|        |       |        |          |        |         |         |        |
|  4975e39d               |  4975e39d               |        |       |        |          |        |         |         |        |
|activemq.management.44065|activemq.management.44065|   1    |   1   |   1    |    1     |   0    |    0    |MULTICAST| false  |
|  c77-addc-410b-8270-e996|  c77-addc-410b-8270-e996|        |       |        |          |        |         |         |        |
|  9dc830b2               |  9dc830b2               |        |       |        |          |        |         |         |        |
|activemq.management.98058|activemq.management.98058|   1    |   0   |   0    |    0     |   0    |    0    |MULTICAST| false  |
|  71f-0572-4664-953c-bb2c|  71f-0572-4664-953c-bb2c|        |       |        |          |        |         |         |        |
|  4f5a1842               |  4f5a1842               |        |       |        |          |        |         |         |        |
```

**Pull request merged**